src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlConnectionPoolKey.cs

```diff
@@ -63,7 +63,7 @@ internal string AccessToken
         public override bool Equals(object obj)
         {
             SqlConnectionPoolKey key = obj as SqlConnectionPoolKey;
-            return (key != null && _credential == key._credential && ConnectionString == key.ConnectionString && Object.ReferenceEquals(_accessToken, key._accessToken));
+            return (key != null && _credential == key._credential && ConnectionString == key.ConnectionString && _accessToken == key._accessToken);
```

Comment:
Can you do a ordinal comparison here for AccessToken ?

Reply:
Sure, I just kept it similar to other comparisons.
Should I change that for Access Token only or for other properties too?

Reply:
Access Token only.